// generator-jhipster/dist/generators/base/shared-data.mjs
export default class SharedData {
    _storage;
    constructor(storage, initialControl = {}) {
        if (!storage) {
            throw new Error('Storage is required for SharedData');
        }
        // Backward compatibility sharedData
        this._storage = storage;
        this._storage.sharedDeployment = this._storage.sharedDeployment || {};
        this._storage.sharedWorkspaces = this._storage.sharedWorkspaces || {};
        this._storage.sharedEntities = this._storage.sharedEntities || {};
        this._storage.sharedApplication = this._storage.sharedApplication || {
            nodeDependencies: {},
        };
        this._storage.sharedSource = this._storage.sharedSource || {};
        this._storage.control = this._storage.control || initialControl;
        this._storage.props = this._storage.props ?? { configOptions: {} };
    }
    getSource() {
        return this._storage.sharedSource;
    }
    getControl() {
        return this._storage.control;
    }
    getApplication() {
        if (!this._storage.sharedApplication)
            throw new Error('Shared application not loaded');
        return this._storage.sharedApplication;
    }
    setEntity(entityName, entity) {
        this._storage.sharedEntities[entityName] = entity;
    }
    hasEntity(entityName) {
        return Boolean(this._storage.sharedEntities[entityName]);
    }
    getEntity(entityName) {
        const entity = this._storage.sharedEntities[entityName];
        if (!entity) {
            throw new Error(`Entity definition not loaded for ${entityName}`);
        }
        return entity;
    }
    getEntities(entityNames = Object.keys(this._storage.sharedEntities)) {
        return entityNames.map(entityName => ({ entityName, entity: this.getEntity(entityName) }));
    }
    getEntitiesMap() {
        return this._storage.sharedEntities;
    }
}
